exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/DrillFileSystem.java

```diff
@@ -89,22 +89,36 @@ public DrillFileSystem(Configuration fsConf) throws IOException {
   }
 
   public DrillFileSystem(Configuration fsConf, OperatorStats operatorStats) throws IOException {
-    this.underlyingFs = FileSystem.get(fsConf);
+    this(fsConf, URI.create(fsConf.getRaw(FS_DEFAULT_NAME_KEY)), operatorStats);
+  }
+
+  public DrillFileSystem(Configuration fsConf, URI Uri, OperatorStats operatorStats) throws IOException {
+    this.underlyingFs = FileSystem.get(Uri, fsConf);
+    logger.trace("Configuration for the DrillFileSystem " + fsConf.getRaw(FS_DEFAULT_NAME_KEY) +
+        ", underlyingFs: " + this.underlyingFs.getUri());
     this.codecFactory = new CompressionCodecFactory(fsConf);
     this.operatorStats = operatorStats;
+    setConf(fsConf);
   }
 
   @Override
   public void setConf(Configuration conf) {
     // Guard against setConf(null) call that is called as part of superclass constructor (Configured) of the
     // DrillFileSystem, at which point underlyingFs is null.
-    if (conf != null && underlyingFs != null) {
-      underlyingFs.setConf(conf);
+    if(conf != null) {
+      super.setConf(conf);
+      if (underlyingFs != null && underlyingFs.getConf() == null) {
+        underlyingFs.setConf(conf);
+      }
+
     }
   }
 
   @Override
   public Configuration getConf() {
+    if (super.getConf() != null) {
```

Comment:
Shouldn't the base class config and the underlying FS config be the same? If they differ, how do we have any idea which is the correct one? In what situation would we want the base class config but not the underlying FS config?

Reply:
No idea, this is extremely confusing. I will just return the conf on the underlying file system. I'll update the other PR if I observe any regressions when I do this.